src/main/java/de/tum/in/www1/artemis/service/connectors/localci/LocalCIBuildJobExecutionService.java

```diff
@@ -171,6 +177,11 @@ public LocalCIBuildResult runBuildJob(ProgrammingExerciseParticipation participa
             throw new LocalCIException("Error while getting branch of participation", e);
         }
 
+        if (commitHash != null && !isTestPush) {
+            // Clone the assignment repository into a temporary directory with the name of the commit hash and then checkout the commit hash.
+            assignmentRepositoryPath = cloneAndCheckoutRepository(participation, commitHash);
+        }
```

Comment:
I think it will be hard to understand why we have this condition, below is how I would solve it, other ideas are appreciated
```suggestion
        /**
        * for test repositories the commit hash of the assignment repository will be relevant
        */
        boolean commitHashFromOtherRepositoryNeeded = commitHash != null && !isTestPush;
        if (commitHash != null && !isTestPush) {
            // Clone the assignment repository into a temporary directory with the name of the commit hash and then checkout the commit hash.
            assignmentRepositoryPath = cloneAndCheckoutRepository(participation, commitHash);
        }
```

Reply:
done